docs/modules/mobile/reconnection-logic.md

```diff
@@ -0,0 +1,71 @@
+
+#### 1. The device OS closes the application.
+
+If the application has been closed, the user has to restart the application. The user will be brought back to the start screen the next time the application is opened and from there on, the application can connect and load any data necessary.
+
+#### 2. The client stops sending data. 
+
+This scenario is a bit more tricky. The application is still alive, however the OS may decide to not run the game or send any data while the application is in the background. [SpatialOS]({{urlRoot}}/reference/glossary#spatialos-runtime) will close the connection, if it doesn't receive any messages from a client-worker in a while. You need to add additional reconnection logic to your game to be able to handle this scenario. 
+
+In the end this is a very game-specific question that depends entirely on what you want to offer to your users. The FPS Starter Project implements the simplest solution: a disconnect takes you back to the start screen, from where you can reconnect to the game.
+
+## How to detect a disconnect
+
+[Heartbeating]({{urlRoot}}/modules/player-lifecycle/heartbeating) is a technique used to verify that your client is still connected. If there are too many failed heartbeats, there are two ways a client-worker may be seen as disconnected:
+
+### 1. SpatialOS believes you disconnected
+
+To ensure that a worker is still connected, the worker has to send messages to SpatialOS every now and thne. This indicates that the worker is still alive. If SpatialOS doesn't receive any message for a while, it will close the connection to that worker. 
+Whenever the connection gets closed on a worker, that worker receives a `DisconnectOp` object. It contains the reason behind the disconnection and triggers a disconnect event in the GDK.
+
+You can listen for this event using the `Worker` object inside the `WorkerConnector` class in order to perform any kind of disconnection logic necessary.
+
+```csharp
+namespace YourGame
+{
+    public class MobileClientWorkerConnector : WorkerConnector
+    {
+    	...
+
+        protected override void HandleWorkerConnectionEstablished()
+        {
+             Worker.OnDisconnect += OnDisconnected;
+        }
+
+        private void OnDisconnected(string reason)
+        {
+            Worker.LogDispatcher.HandleLog(LogType.Log, new LogEvent($"Worker disconnected")
+                .WithField("WorkerId", Worker.WorkerId)
+                .WithField("Reason", reason));
+        }
+    }
+}
+```
+
+### 2. The player lifecycle module believes you disconnected 
+
+Our [Player Lifecycle module]({{urlRoot}}/modules/player-lifecycle/overview) allows you to easily create players and also implements heartbeating to delete the player entity of any unresponsive client-workers. In mobile applications, this can result in entities being destroyed even though the worker is still connected to SpatialOS. If this happens, that client-worker won't receive a `DisconnectOp`.
```

Comment:
```suggestion
Our [Player Lifecycle module]({{urlRoot}}/modules/player-lifecycle/overview) allows you to easily create players and uses the heartbeating technique to detect and delete the player entity of any unresponsive client-workers. In mobile applications, this can result in entities being destroyed even though the worker is still connected to SpatialOS. If this happens, that client-worker won't receive a `DisconnectOp`.
```

Reply:
> In mobile applications, this can result in entities being destroyed even though the worker is still connected to SpatialOS.

Under what conditions can this happen? I.e. - client level heartbeats fail, but the underlying SpatialOS ones don't

Reply:
yeah, will add a sentence about that